### PR TITLE
Feat/tools

### PR DESCRIPTION
**Pull request by tomasliu-agora**
Add Tool Support for openai realtime api

### CODE REVIEW

File: agents/property.json [hunk 1]
```diff
@@ -2253,6 +2277,162 @@
...
+                                        "extension": "agora_rtc"
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "va.openai.v2v.tools",
```
Comment (wangyoucao577) [inline agents/property.json:2310]: So this is not necessary anymore and can be removed, right?

File: agents/ten_packages/extension/openai_v2v_python/client.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        self.max_tokens = max_tokens
+        self.voice = voice
+        self.server_vad = server_vad
+    
+    def build_ctx(self) -> dict:
+        return {
+            "language": self.language
+        }
+
+class RealtimeApiClient:
```
Comment (wangyoucao577) [inline agents/ten_packages/extension/openai_v2v_python/client.py:73]: It seems somehow duplicated with the `realtime/connection.py`?

Comment (tomasliu-agora) [reply agents/ten_packages/extension/openai_v2v_python/client.py]: removed

File: agents/property.json (hunk 1 above)
Comment (tomasliu-agora) [reply agents/property.json]: done

**Review by wangyoucao577 — APPROVED** [review]

**Review by plutoless — CHANGES_REQUESTED** [review]

**Review by plutoless — APPROVED** [review]

File: agents/ten_packages/extension/openai_v2v_python/tools.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    "description": "The location or region for the weather check.",
+                },
+                "datetime": {
+                    "type": "string",
+                    "description": "The date and time for the weather check. The datetime should use format like 2024-10-01T16:42:00.",
+                }
+            },
+            "required": ["location"],
+        })
+    print(r.to_prompt())
```
[the remark this answers is not in the document]
Comment (tomasliu-agora) [reply agents/ten_packages/extension/openai_v2v_python/tools.py:84]: this is test script.

**Pull request merged**